DNN Platform/Dnn.AuthServices.Jwt/Services/MobileHelperController.cs

```diff
@@ -0,0 +1,155 @@
+                    {
+                        TabId = tabmodule.TabInfo.TabID,
+                        ModuleId = tabmodule.ModuleInfo.ModuleID,
+                        PageName = tabmodule.TabInfo.TabName,
+                        PagePath = tabmodule.TabInfo.TabPath
+                    });
+                    siteDetails.Modules.Add(moduleDetail);
+                }
+            }
+
+            return Request.CreateResponse(HttpStatusCode.OK, siteDetails);
+        }
+
+        #region private methods
+
+        private static IEnumerable<TabModule> GetTabModules(string moduleName)
+        {
+            var portalId = PortalController.Instance.GetCurrentPortalSettings().PortalId;
+            var desktopModule = DesktopModuleController.GetDesktopModuleByModuleName(moduleName, portalId);
+            if (desktopModule != null)
+            {
+
+                var cacheKey = string.Format(DataCache.DesktopModuleCacheKey, portalId) + "_" +
+                               desktopModule.DesktopModuleID;
+                var args = new CacheItemArgs(cacheKey, DataCache.DesktopModuleCacheTimeOut,
+                                             DataCache.DesktopModuleCachePriority, portalId, desktopModule);
+
+                return CBO.GetCachedObject<IList<TabModule>>(args, GetTabModulesCallback);
+            }
+
+            return new List<TabModule>();
+        }
+
+        private static object GetTabModulesCallback(CacheItemArgs cacheItemArgs)
+        {
+            var tabModules = new List<TabModule>();
+
+            var portalId = (int)cacheItemArgs.ParamList[0];
+            var desktopModule = (DesktopModuleInfo)cacheItemArgs.ParamList[1];
+
+            var tabController = new TabController();
+            var tabsWithModule = tabController.GetTabsByPackageID(portalId, desktopModule.PackageID, false);
+            var allPortalTabs = tabController.GetTabsByPortal(portalId);
+            IDictionary<int, TabInfo> tabsInOrder = new Dictionary<int, TabInfo>();
+
+            //must get each tab, they parent may not exist
+            foreach (var tab in allPortalTabs.Values)
+            {
+                AddChildTabsToList(tab, ref allPortalTabs, ref tabsWithModule, ref tabsInOrder);
+            }
+
+            foreach (var tab in tabsInOrder.Values)
+            {
+                tabModules.AddRange(
+                    tab.ChildModules.Values.Where(
+                        childModule => childModule.DesktopModuleID == desktopModule.DesktopModuleID)
+                       .Select(childModule => new TabModule
+                       {
+                           TabInfo = tab,
+                           ModuleInfo = childModule,
+                           ModuleVersion = desktopModule.Version
+                       }));
+            }
+
+            return tabModules;
+        }
+
+        private static void AddChildTabsToList(TabInfo currentTab, ref TabCollection allPortalTabs,
```

Comment:
There shouldn't be a need to pass `TabCollection` or `IDictionary` by ref, since they're reference types already